### PR TITLE
Remove support for jextract -interactive

### PR DESCRIPTION
**Pull request by TheMarvelFan**
Referencing issue #10764 
I am not sure how to test this change on my local environment, but I am sure that its support has been ended. Please do point out any error or some additional change that needs to be made and I might have overlooked, and feel free to leave reviews about this contribution. PS: I am new to open source contribution and this will be my first actual pull request, so please provide guidance if I have made any errors.

### CODE REVIEW

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/JextractCommand.java [hunk 1]
```diff
@@ -44,4 +44,4 @@ public void run(String command, String[] args, Context context, PrintStream out)
 		}
 	}
 
-}
+}
```
Comment (keithc-ca) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/JextractCommand.java:47]: We want to keep the final newline.
The only other changes I see are removal of trailing whitespace.
Please restore the final newline or just revert changes to this file.

File: debugtools/DDR_VM/testsrc/com/ibm/j9ddr/tools/xml/jextract/JExtractXMLGenerator.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-/*******************************************************************************
+	/*******************************************************************************
```
Comment (keithc-ca) [inline debugtools/DDR_VM/testsrc/com/ibm/j9ddr/tools/xml/jextract/JExtractXMLGenerator.java:1]: Please undo this change.

File: jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java [hunk 3]
```diff
@@ -338,14 +335,10 @@ public static void main(String[] args) {
 
 		Main dumper = new Main(dumpName, virtualRootDirectory, verbose, throwExceptions, disableBuildIdCheck, excludeCoreFile);
 
-		if (interactive) {
-			dumper.runInteractive();
-		} else {
 			if (zip) {
 				dumper.runZip(outputName);
 			}
 			report("jextract complete."); //$NON-NLS-1$
```
Comment (keithc-ca) [inline jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java:341]: Please also remove `zip` (which can only ever be `true`), and correct the indentation here.

**Review by keithc-ca — CHANGES_REQUESTED** [review]
Please remove the (now) unused code:
* Main.runInteractive()
* supporting imports
* Main.doCommand() and associated native code, exports, etc. (jextraceglue.c, jextractnatives_internal.h, jextractnatives.c, jextractnatives_api.h, module.xml, CMakeLists.txt)

**Comment by keithc-ca** [comment]
> how to test this change

See https://github.com/eclipse-openj9/openj9/blob/master/doc/build-instructions/Build_Instructions_V21.md for how to set up your system and build (Java 21 in this case) locally.

**Comment by TheMarvelFan** [comment]
@keithc-ca Thanks for pointing these errors out. I am working on the exports, and have made the changes you mentioned in the Java source code.

File: jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java [hunk 4]
```diff
@@ -45,7 +43,6 @@
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipOutputStream;
 
-import com.ibm.dtfj.addressspace.IAbstractAddressSpace;
```
Comment (keithc-ca) [inline jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java:48]: This import is required: please restore it.

File: jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java [hunk 5]
```diff
@@ -338,14 +331,8 @@ public static void main(String[] args) {
...
 		Main dumper = new Main(dumpName, virtualRootDirectory, verbose, throwExceptions, disableBuildIdCheck, excludeCoreFile);
 
-		if (interactive) {
-			dumper.runInteractive();
-		} else {
-			if (zip) {
-				dumper.runZip(outputName);
-			}
+			dumper.runZip(outputName);
 			report("jextract complete."); //$NON-NLS-1$
```
Comment (keithc-ca) [inline jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java:335]: Please correct the indentation here (one tab too many on lines 334 and 335).

File: jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java [hunk 6]
```diff
@@ -676,8 +628,4 @@ private static void copy(ClosingFileReader from, OutputStream to, byte[] buffer)
 		}
 	}
 
-	private native long getEnvironmentPointer(IAbstractAddressSpace dump, boolean disableBuildIdCheck) throws Exception;
```
Comment (keithc-ca) [inline jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java:679]: This method is still used; please restore it.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java [hunk 7]
```diff
@@ -45,11 +43,11 @@
 import java.util.zip.ZipEntry;
 import java.util.zip.ZipOutputStream;
 
-import com.ibm.dtfj.addressspace.IAbstractAddressSpace;
 import com.ibm.dtfj.corereaders.Builder;
 import com.ibm.dtfj.corereaders.ClosingFileReader;
 import com.ibm.dtfj.corereaders.DumpFactory;
 import com.ibm.dtfj.corereaders.ICoreFileReader;
+import com.ibm.dtfj.addressspace.IAbstractAddressSpace;
```
Comment (keithc-ca) [inline jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java:50]: Please keep the import list sorted (i.e. move this before line 49).

File: runtime/oti/jextractnatives_api.h [hunk 8]
```diff
@@ -43,18 +43,6 @@ extern "C" {
 #endif
 
 /* ---------------- jextractnatives.c ---------------- */
-
```
Comment (keithc-ca) [inline runtime/oti/jextractnatives_api.h:46]: Please leave one blank line after this comment.

File: runtime/oti/jextractnatives_api.h [hunk 9]
```diff
@@ -43,18 +43,6 @@ extern "C" {
...
-/**
-* @brief
-* @param *env
-* @param obj
-* @param dumpObj
-* @param commandObject
-* @return void
-*/
-void JNICALL
-Java_com_ibm_jvm_j9_dump_extract_Main_doCommand(JNIEnv *env, jobject obj, jobject dumpObj, jstring commandObject);
```
Comment (keithc-ca) [inline runtime/oti/jextractnatives_api.h:56]: Please also remove the definition in `jextractnatives.c`, and the exports in `runtime/jextractnatives/CMakeLists.txt` and `runtime/jextractnatives/module.xml`.

That eliminates the only use of `void run_command(const char *)` (in `runtime/jextractnatives/jextractglue.c`). Since that source file contains no other definitions, it and `jextractnatives_internal.h` can be removed.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

**Comment by TheMarvelFan** [comment]
@keithc-ca Thanks for the reviews. I hope everything is in order now.

File: jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java (hunk 7 above)
Comment (keithc-ca) [reply jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java]: Perhaps a line number wasn't the best way to explain where it belongs. It should appear before `Builder`:
```java
import com.ibm.dtfj.addressspace.IAbstractAddressSpace;
import com.ibm.dtfj.corereaders.Builder;
```

File: jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java (hunk 5 above)
Comment (keithc-ca) [reply jcl/src/openj9.dtfj/share/classes/com/ibm/jvm/j9/dump/extract/Main.java:336]: These two lines are still indented (one tab) too much.

File: runtime/jextractnatives/CMakeLists.txt [hunk 10]
```diff
@@ -39,7 +39,6 @@ target_link_libraries(j9jextract
 )
 
 omr_add_exports(j9jextract
-	Java_com_ibm_jvm_j9_dump_extract_Main_doCommand
```
Comment (keithc-ca) [inline runtime/jextractnatives/CMakeLists.txt:42]: Line 26 must be deleted as it refers to a file that is being deleted.
While you're here, please also delete (mostly blank) line 28.

File: runtime/jextractnatives/jextractnatives.c [hunk 11]
```diff
@@ -289,29 +289,6 @@ cacheIDs(JNIEnv* env, jobject dumpObj)
 	return 0;
 }
 
-void JNICALL
```
Comment (keithc-ca) [inline runtime/jextractnatives/jextractnatives.c:292]: Line 28
```c
#include "jextractnatives_internal.h"
```
must be deleted; it refers to a file deleted by this change.

Also, please don't leave two consecutive blank lines here.

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: runtime/jextractnatives/CMakeLists.txt (hunk 10 above)
Comment (TheMarvelFan) [reply runtime/jextractnatives/CMakeLists.txt:42]: Shall I also remove line 34 as it is mostly blank too?

Comment (keithc-ca) [reply runtime/jextractnatives/CMakeLists.txt:42]: Yes, I think that's a good idea.

**Comment by keithc-ca** [comment]
The overall change looks good. I think it's time to rebase and squash to a single commit.

**Comment by TheMarvelFan** [comment]
@keithc-ca Hi Keith. I have squashed all the commits into one. Thanks a lot for your cooperation and patience, and apologies for any trouble I may have caused. I learnt a lot of new things while I contributed to this issue, and I will put my knowledge to good use.

**Comment by keithc-ca** [comment]
Thank you for squashing to a single commit. It would be nice if you could rebase on a more recent commit on the master branch, but that's less important.

I would like you to improve the commit message; see [1] for guidance. The first line should summarize what merging the commit will do (the summary here, "Remove support for jextract -interactive", would be fine). Normally, the body would provide more detail, but I'm not sure there's much more to say. On the other hand, I don't think it's helpful to have text like "removed blank line" or "removed another extra line", etc.

[1] https://github.com/eclipse-openj9/openj9/blob/master/CONTRIBUTING.md

**Comment by keithc-ca** [comment]
First, a minor nit: the first line of the commit message should start with "Remove", not "Removed".

I'm not sure where the other 19 commits came from, but I think what you want to do is
```sh
git fetch --all
git rebase -i 5570f22bf9da
```
This will open an editor showing
```
pick 29d12dea56a Add tests for Value Type System.arraycopy transformation
pick bb5ffa41a84 Removed support for jextract -interactive

# other text with instructions...
```
You want to change "pick" to "drop" on the first line. Save the file and exit the editor. Git will skip commit 29d12dea56a and create a new commit with the other changes.

You'll want to remove what I expect is some intermediate temporary file
```sh
git rm runtime/compiler/build/scripts/RCa05068
```
and then amend that commit
```
git commit --amend
```
Again, git will open an editor allowing you to update the (amended) commit message; here's your change to remove that unwanted "d".

**Comment by TheMarvelFan** [comment]
@keithc-ca Apologies for those extra commits being added. I pushed to origin right after fetching master by mistake, and that might be what caused those extra commits to show. Thanks for the help.

**Review by keithc-ca — APPROVED** [review]

**Comment by keithc-ca** [comment]
Jenkins compile aix,alinux,win jdk17

**Comment by keithc-ca** [comment]
Jenkins test sanity amac jdk11
